src/karma.js

```diff
@@ -109,16 +109,21 @@ export default class KarmaClient extends EventEmitter {
 	}
 
 	applyShims(scriptUrl) {
-		if (scriptUrl.indexOf('mocha') !== -1) {
+		if (scriptUrl.includes('mocha')) {
 			global.window = global;
 			global.location = {
 				pathname: '/'
 			};
+		} else if (scriptUrl.includes('jasmine')) {
+			global.window = global;
+			global.location = {
+				origin: 'null'
+			};
 		}
 	}
 
 	removeShims(scriptUrl) {
-		if (scriptUrl.indexOf('mocha') !== -1) {
+		if (scriptUrl.includes('mocha')) {
```

Comment:
Do you need to remove the shim for jasmine here as well?

Reply:
No, it needs to be available during runtime later on. A part of the shim for mocha is only required while the mocha adapter is initializing, that's why it can be removed after the script run.